### ISSUE TITLE
Data loss when editing symlinked project

### ISSUE THREAD

**Issue by tbenst** [atom/atom#17191]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
Atom fails to use proper filepath when a project is opened in a symlinked file and a file is opened using "Find File"

### Steps to Reproduce
1. ```
$ mkdir project
$ ln -s project linktoproject
$ cd linktoproject
$ echo hello > hi.txt
$ atom .
```
2. Click `hi.txt` on project sidebar. Make some edits without saving
3. `c-p` for find file and open `hi.txt`. Make changes and save
4. go back to first tab and save

**Expected behavior:**
I should have two views of the same file, with edits propagating to both. `c-p` should open `linktoproject/hi.txt`

**Actual behavior:**
Atom opens `linktoproject/hi.txt` on click but opens `project/hi.txt` from find file. Atom thinks these are two different files. After saving both files, there is no indication that one of the files now has unsaved information that will be forever lost if closed.

**Reproduces how often:** [What percentage of the time does it reproduce?]
100%

### Versions

You can get this information from copy and pasting the output of `atom --version` and `apm --version` from the command line. Also, please include the OS and what version of the OS you're running.
Atom    : 1.23.3
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0

apm  1.18.12
npm  3.10.10
node 6.9.5 x64
atom 1.23.3
python 3.6.2
git 2.7.4


### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by rsese** [comment]
Thanks for the report!

> Atom : 1.23.3

Can you update to the latest release version (currently 1.26.0) and check if you still see this behavior?  Also, can you confirm that you installed Atom with one of the supported [installation methods](https://flight-manual.atom.io/getting-started/sections/installing-atom/)?

**Comment by tbenst** [comment]
Thanks for the response. I updated using the apt repo per your link and can confirm that the problem remains in 1.26.0

**Comment by rsese** [comment]
Do you see any errors in devtools console?

https://flight-manual.atom.io/hacking-atom/sections/debugging/#check-for-errors-in-the-developer-tools

I'll get my Ubuntu vm running to try and reproduce.

**Comment by rsese** [comment]
Ohhh sorry I see what you're saying now - since you have the file opened twice as you described, the behavior seems the same as described in https://github.com/atom/atom/issues/3594?

I'm not sure if the fuzzy finder behavior is expected though (it behaves the same with `Settings > Core > Follow Symlinks` enabled and disabled).

**Comment by tbenst** [comment]
Yes, https://github.com/atom/atom/issues/3594 seems related to the problem, although in my example only Atom is editing the file. I also think that the fuzzy finder is opening the wrong path

**Comment by tbenst** [comment]
This issue just struck me again today--happy to try to help if you can point me somewhere in the code base?

**Comment by Azzam-sa4** [comment]
> أدهشتني هذه المشكلة للتو مرة أخرى اليوم - هل يسعدني أن أحاول مساعدتك إذا كان بإمكانك توجيهي إلى مكان ما في قاعدة التعليمات البرمجية؟

Yes pls..
